mbuild/tests/test_lammpsdata.py

```diff
@@ -47,12 +47,96 @@ def test_save_charmm(self):
                     in out_lammps[i + 1]
                 )
                 assert len(out_lammps[i + 2].split("#")[0].split()) == 5
+                found_angles = True
+            elif "Dihedral Coeffs" in line:
+                assert "# charmm" in line
+                assert "#k, n, phi, weight" in out_lammps[i + 1]
```

Comment:
For at least the two conditions for the weighting factor we added, can you add in a test that verifies that the weighting factor is being set to zero?

Reply:
The assertion that the weighting factor is zero when `zero_dihedral_weighting_factor = True` was already in `test_amber`.  I just added an assertion that the weighting factor is 1.0 in `test_save_charmm`, where `zero_dihedral_weighting_factor` is left at the default, which is `False`.

Reply:
Unless I’m misunderstanding, if the writhing factor is only 1 or 0, do we need to use the list length?  Or do we just set to 1 or 0?

Reply:
@bc118 When `zero_dihedral_weighting_factor = False`, the weighting factor is 1.0/(number of terms in the dihedral), which can be 1.0, 0.5, 0.3333, etc.

Reply:
Should we add a test where it is 1/2, like in a cyclic ring?  I’m ok with not doing it if else is.

Reply:
When the method of determining the weighting factor for CHARMM dihedrals is fixed in a future PR, it should include such a test, but the current method has nothing to do with correcting for rings (which is the actual intended purpose of the weighting factor).  Any test made now in which it is 0.5 would probably break when that fix is eventually made.

Reply:
Ok.  Thanks for clarifying.  I’m approving then